### PR TITLE
Chore icu 9958 storybook patch babel v7 mode

### PR DESCRIPTION
**Pull request by calcaide**
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-9958)

## Description

- Update `@storybook/*` to `6.4.x`.
- After the updates, we re-audit specific resoltions for `@storybook`. Resulting in a bunch of them being deleted.
- Missing migrate storybook to babel mode v7, as mentioned in the Jira ticket. We will attempt this later when upgrading other deps because we run into dependency issues when attempted.

## How to test this PR:

After every commit, the test procedure it's been building and running `@storybook` succesfully.

### CODE REVIEW

**Review by ZedLi — APPROVED** [review]
This is fantastic!! Thanks for doing this upgrade work!

File: addons/rose/package.json [hunk 1]
```diff
@@ -57,18 +57,20 @@
...
+    "@storybook/addon-actions": "^6.4.0",
     "@storybook/addon-controls": "^6.5.12",
-    "@storybook/addon-docs": "~6.3.8",
-    "@storybook/addons": "^6.3.9",
-    "@storybook/api": "^6.3.12",
-    "@storybook/cli": "^6.3.10",
-    "@storybook/components": "^6.3.12",
-    "@storybook/core-events": "^6.3.0",
-    "@storybook/ember": "6.4.0",
+    "@storybook/addon-docs": "~6.4.0",
```
[the remark this answers is not in the document]
Comment (calcaide) [reply addons/rose/package.json:63]: @ZedLi 

Awesome questions!!

`@storybook/addons-docs` is getting important changes/deprecations within `6.5`, [here more information](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#from-version-64x-to-650) about it. So to minimize noise within this PR I was trying to keep it pined with `~`. But definitely planning on deprecation work and upgrade `@storybook/addons-doc` to `6.5`.

---

Absolutely agreed with your second comment.

Upgrading storybook it's hard and easy to get lost within dependencies of dependencies. 
After trying different strategies without much success, pining the main dependencies, such as `@storybook/ember`, and troublesome dependencies, such as `@storybook/docs` while being more open with `^` in the rest of them, brings less bugs/noise to validate the entire storybook works after an upgrade. Also facilitates deprecation work.

Then, as you mention, we resolve higher versions than we specify, i.e: we declare `@storybook/depA: ^6.4.0` and we resolve `@storybook/depA@6.5.24`. But re-declaring those, sometimes ends in unexpected bugs/noise and loops of frustration. 

In a nutshell, I am trying to "box" the storybook upgrades efforts in a way that we do baby steps = PR, validate it, and move to the next step (which will be upgrade to `^6.5.x` `@storybook` deps).

Happy to discuss this over slack of zoom 😉

**Review by DhariniJeeva — APPROVED** [review]
